Lib/pickle.py

```diff
@@ -314,16 +314,17 @@ def load_frame(self, frame_size):
 # Tools used for pickling.
 
 def _getattribute(obj, name):
+    top = obj
     for subpath in name.split('.'):
         if subpath == '<locals>':
             raise AttributeError("Can't get local attribute {!r} on {!r}"
-                                 .format(name, obj))
+                                 .format(name, top))
```

Comment:
Could we take the chance of using f-strings instead?

Reply:
Maybe in other PRs. I am going to significantly rewrite this code, so using f-strings will be a minor part.